recipes/asio/all/conanfile.py

```diff
@@ -1,38 +1,47 @@
-from conans import ConanFile, tools
+from conan import ConanFile
+from conan.tools.files import copy, get
+from conan.tools.layout import basic_layout
 import os
 
-required_conan_version = ">=1.33.0"
+required_conan_version = ">=1.50.0"
 
 
 class Asio(ConanFile):
     name = "asio"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "http://think-async.com/Asio"
     description = "Asio is a cross-platform C++ library for network and low-level I/O"
-    topics = ("conan", "asio", "network", "io", "low-level")
-    settings = "os"
+    topics = ("asio", "network", "io", "low-level")
     license = "BSL-1.0"
+    settings = "os", "arch", "compiler", "build_type"
     no_copy_source = True
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
     def package_id(self):
-        self.info.header_only()
+        self.info.clear()
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def build(self):
+        pass
 
     def package(self):
-        root_dir = os.path.join(self._source_subfolder, self.name)
+        root_dir = os.path.join(self.source_folder, "asio")
         include_dir = os.path.join(root_dir, "include")
-        self.copy(pattern="LICENSE_1_0.txt", dst="licenses", src=root_dir)
-        self.copy(pattern="*.hpp", dst="include", src=include_dir)
-        self.copy(pattern="*.ipp", dst="include", src=include_dir)
+        copy(self, "LICENSE_1_0.txt", src=root_dir, dst=os.path.join(self.package_folder, "licenses"))
+        copy(self, "*.hpp", src=include_dir, dst=os.path.join(self.package_folder, "include"))
+        copy(self, "*.ipp", src=include_dir, dst=os.path.join(self.package_folder, "include"))
 
     def package_info(self):
+        self.cpp_info.set_property("pkg_config_name", "asio")
```

Comment:
Where does this come from 🤔 did know there was individual ones

Reply:
https://github.com/chriskohlhoff/asio/blob/master/asio/Makefile.am#L3

Reply:
🤢 there's two repos?! https://github.com/boostorg/asio 😕